src/App.tsx

```diff
@@ -4,57 +4,110 @@
+      .then(setUserPosts)
+      .catch(() => setHasError(true))
+      .finally(() => setIsPostsLoading(false));
+  }, [selectedUser]);
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
+  return (
+    <main className="section">
+      <div className="container">
+        <div className="tile is-ancestor">
+          <div className="tile is-parent">
+            <div className="tile is-child box is-success">
+              <div className="block">
+                <UserSelector
+                  users={users}
+                  selectedUser={selectedUser}
+                  setSelectedUser={setSelectedUser}
+                />
               </div>
 
-              <div className="notification is-warning" data-cy="NoPostsYet">
-                No posts yet
-              </div>
+              {isPostsLoading ? (
+                <Loader />
+              ) : (
+                <div className="block" data-cy="MainContent">
+                  {!selectedUser && (
+                    <p data-cy="NoSelectedUser">No user selected</p>
+                  )}
+
+                  {hasError && !isPostsLoading && (
+                    <div
+                      className="notification is-danger"
+                      data-cy="PostsLoadingError"
+                    >
+                      Something went wrong!
+                    </div>
+                  )}
 
-              <PostsList />
+                  {selectedUser &&
+                    !hasError &&
+                    (!userPosts.length ? (
+                      <div
+                        className="notification is-warning"
+                        data-cy="NoPostsYet"
+                      >
+                        No posts yet
+                      </div>
+                    ) : (
+                      <PostsList
+                        userPosts={userPosts}
+                        selectedPost={selectedPost}
+                        setSelectedPost={setSelectedPost}
+                      />
+                    ))}
+                </div>
+              )}
             </div>
           </div>
-        </div>
 
-        <div
-          data-cy="Sidebar"
-          className={classNames(
-            'tile',
-            'is-parent',
-            'is-8-desktop',
-            'Sidebar',
-            'Sidebar--open',
-          )}
-        >
-          <div className="tile is-child box is-success ">
-            <PostDetails />
+          <div
+            data-cy="Sidebar"
+            className={classNames(
+              'tile',
+              'is-parent',
+              'is-8-desktop',
+              'Sidebar',
+              { 'Sidebar--open': selectedPost !== null },
+            )}
+          >
+            {selectedPost && (
+              <div className="tile is-child box is-success ">
+                <PostDetails selestedPost={selectedPost} />
```

Comment:
There is a typo in the `PostDetails` component prop. It should be `selectedPost` instead of `selestedPost`.